### ISSUE TITLE
Wrong reasults after Tuning

### ISSUE THREAD

**Issue by SamihaSara** [VisualComputingInstitute/TrackR-CNN#79]
I followed the steps in README.md. But after tuning, I am getting sMOTSA, MOTSA, IDS etc values zero.  Below are the Configurations--->

1)  I set num_epochs=1 and In config files wrote "load_init": "/TrackR-CNN/models/converted/". 

models
└──>
           ── /conv3d_sep2/(conv3d_sep2-00000005.data-00000-of-00001,  conv3d_sep2- 
                                                                                           00000005.index, conv3d_sep2-00000005.meta), 
           ──/converted/(converted.data-00000-of-00001 , converted.meta and converted.index)

Is this how files should be arranged in /models/?

2) After Forwarding and tracking, In both  /forwarded/conv3d_sep2/detections/5/ and /forwarded/conv3d_sep2/tracking_data/
folders, txt files are 0 bytes. /forwarded/conv3d_sep2/vis/detections/ has wrong identifications (it did not identified cars and peds) and conv3d_sep2/vis/tracks/ had the same as input images. Is  /path/to/output_file a directory?

3) Finally, after tuning I got the evaluation matrix zero. with path/to/precomputed_optical_flow/ directory keeping empty. Files in 
/path/to/tmp_folder/ are also of 0 byte.

Please let me know how I can solve this issue to get the correct results. Also, if I want to set association_type to mask or bbox_iou then what should be in /path/to/precomputed_optical_flow?

**Comment by pvoigtlaender** [comment]
Hi,

please first try to get a non-empty output for forwarding, otherwise trying tuning/optical flow does not make any sense.
From your description it's not easy to see what's happened, probably some problem with paths.
Make sure you have 
`"load": "models/conv3d_sep2-00000005",`
in your config and to set the paths to the data.
Maybe you'll have to use a debugger to find out why all the results are empty.

**Comment by SamihaSara** [comment]
Can you suggest any debugger to analyze the results?

**Comment by pvoigtlaender** [comment]
I recommend using PyCharm as IDE which includes a debugger

**Comment by SamihaSara** [comment]
In another comment, you told not to specify "load". Are "load" and "load_init"  different variables in config files? 
I just tried with the following 

models
└──>
── /conv3d_sep2/(conv3d_sep2-00000005.data-00000-of-00001, conv3d_sep2-00000005.index, conv3d_sep2-00000005.meta),

With "model": "trained_model"  But getting this error even if there are these checkpoint files.

![image](https://user-images.githubusercontent.com/9627710/87683967-86008880-c7a3-11ea-975b-ceaf7839d671.png)

However, this error does not occur if  "model": is not renamed in configs. How can I solve this issue?

**Comment by pvoigtlaender** [comment]
Please try
`"load_init": "models/conv3d_sep2/conv3d_sep2-00000005",`

**Comment by SamihaSara** [comment]
Yes, this worked. Now for renaming "model", I noticed in different config files it had different values.  So should I also rename with different names or "model": "trained_model" this name can be used in all config files?
How much time it may take to train this model?

Before I tried to with 
"load_init": "/home/samiha/TrackR-CNN/models/converted",
which took around 12 hours to train but got Files  in  /forwarded  of 0 byte.  Is it because this COCO+Mapillary pretrained model does not fit with KITTI_MOTS dataset?

**Comment by ahnonay** [comment]
I'm not quite clear about what the problem is. Are you able to get correct forwarding results if you follow the steps in the Readme (i.e. !not! training your own model, but using our trained conv3d_sep2 model)?

**Comment by SamihaSara** [comment]
@ahnonay How much time TrackR-CNN should take if it's trained on KITTI_MOTS dataset? For me, it took 12 hours so I was wondering if it was trained properly. If I use COCO+Mapillary dataset for training, should I change the pre-trained model from conv3d_sep2 to models/converted?

**Comment by ahnonay** [comment]
Training time depends on your hardware, but 12 hours sounds reasonable.
Both the conv3d_sep2 and the "converted" model have already been trained on COCO+Mapillary. If you want to retrain on KITTI_MOTS, you should probably use the "converted" model.

**Comment by SamihaSara** [comment]
@ahnonay But if I use "conv3d_sep2" model, doesn't TrackRCNN retrain on KITTI_MOTS? Can you please provide some details about "converted" and "conv3d_sep2" models? Which architectures do they follow? Also, Can I train TrackRCNN without using "converted"  or "conv3d_sep2"?

**Comment by ahnonay** [comment]
If you use the conv3d_sep2 model without changing the config file, there will be no further training on KITTI_MOTS (since the model has already been trained for the 5 epochs determined in the config - you could increase the number of epochs in the config to train longer).

The "converted" model in trackrcnn_init.zip is a Mask R-CNN pretrained on COCO+Mapillary. The conv3d_sep2 model is a TrackR-CNN initialized with the "converted" model and finetuned on KITTI_MOTS. 

You can train TrackR-CNN from scratch of course. However, KITTI_MOTS is probably too small of a dataset to do that effectively.

**Comment by SamihaSara** [comment]
If I change the config files (by putting "load": "models/conv3d_sep2-00000005") does it train from scratch? Do I need to rename the model name in all the config files?
How do 'converted' and 'conv3d_sep2' effect the tracking results? 
As conv3d_sep2 already finetuned on KITII_MOTS, How does it help as a pre-trained model while training TrackR-CNN on KITTI_MOTS?
To reproduce the paper's result, should it be trained on both MOTS-Challenge and KITTI_MOTS?

**Comment by ahnonay** [comment]
If you put "load": "models/conv3d_sep2-00000005", then it will train from the KITTI_MOTS finetuned checkpoint, *not* from scratch. If you want to train completely from scratch, choose a new model name and do not use any "load" command.

"conv3d_sep2" works well for tracking. "converted" will not work well, since the association head will be initialized randomly.

You can use the "conv3d_sep2" checkpoint and continue training on KITTI_MOTS of course. It might get a little better still.

To reproduce the results in the paper, you can either
- use the "conv3d_sep2" model without any more training (just forwarding and tracking)
- train your own "conv3d_sep2" model using "converted" as pretrained weights (the results will be slightly different than in the paper due to random initialization etc)
